### PR TITLE
fix(datagrid): allow to mutate data in the grid

### PR DESCRIPTION
**Pull request by romainseb**
**What is the problem this PR is trying to solve?**
With the current api, the grid is reset when the data change
**What is the chosen solution to this problem?**
Allow to pass rowData in props to fill the datagrid
Add a loading feedback

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [x] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/datagrid/src/components/DefaultCellRenderer/DefaultCellRenderer.component.js [hunk 1]
```diff
@@ -10,7 +11,15 @@ import theme from './DefaultCell.scss';
 
 export const CELL_RENDERER_COMPONENT = 'cellRenderer';
 
-export default function DefaultCellRenderer({ avroRenderer, colDef, value, getComponent }) {
+export default function DefaultCellRenderer({ avroRenderer, colDef, value, getComponent, data }) {
+	if (data.loading) {
```
Comment (jare-talend) [inline packages/datagrid/src/components/DefaultCellRenderer/DefaultCellRenderer.component.js]: could you add a test on this.

File: packages/datagrid/src/components/DataGrid/DataGrid.component.js [hunk 2]
```diff
@@ -150,6 +158,11 @@ export default class DataGrid extends React.Component {
 	}
 
 	getAgGridConfig() {
+		let rowData = this.props.rowData;
```
Comment (jare-talend) [inline packages/datagrid/src/components/DataGrid/DataGrid.component.js]: This props is used ? If yes, could you add the props on the proptypes ?

File: packages/datagrid/src/components/DefaultCellRenderer/DefaultCellRenderer.component.js (hunk 1 above)
Comment (romainseb) [reply packages/datagrid/src/components/DefaultCellRenderer/DefaultCellRenderer.component.js]: It's done :)

File: packages/datagrid/src/components/DataGrid/DataGrid.component.js (hunk 2 above)
Comment (romainseb) [reply packages/datagrid/src/components/DataGrid/DataGrid.component.js]: It's done :)

**Pull request merged**